capz/run-capz-e2e.sh

```diff
@@ -116,143 +121,44 @@ create_cluster(){
         log "cluster creation complete"
     fi
 
-    # set the kube config to the workload cluster
-    export KUBECONFIG="$PWD"/"${CLUSTER_NAME}".kubeconfig
-}
-
-apply_workload_configuraiton(){
-    # Only patch up kube-proxy if $WINDOWS_KPNG is unset
-    if [[ -z "$KPNG" ]]; then
-        # A patch is needed to tell kube-proxy to use CI binaries.  This could go away once we have build scripts for kubeproxy HostProcess image.
-        kubectl apply -f "${CAPZ_DIR}"/templates/test/ci/patches/windows-kubeproxy-ci.yaml
-        kubectl rollout restart ds -n kube-system kube-proxy-windows
+    # set the kube config to the workload cluster only if tests run on local system
+    if [[ ! "${PRIVATE_TESTING}" == "true" ]]; then
+        export KUBECONFIG="$PWD"/"${CLUSTER_NAME}".kubeconfig
     fi
-
-    # apply additional helper manifests (logger etc)
-    kubectl apply -f "${CAPZ_DIR}"/templates/addons/windows/containerd-logging/containerd-logger.yaml
-    kubectl apply -f "${CAPZ_DIR}"/templates/addons/windows/csi-proxy/csi-proxy.yaml
-    kubectl apply -f "${CAPZ_DIR}"/templates/addons/metrics-server/metrics-server.yaml
 }
 
-apply_hyperv_configuration(){
-    set -x
-    log "applying contirguration for testing hyperv isolated containers"
-
-    log "installing hyperv runtime class"
-    kubectl apply -f "${SCRIPT_ROOT}/../helpers/hyper-v-mutating-webhook/hyperv-runtimeclass.yaml"
-
-    # ensure cert-manager and webhook pods land on Linux nodes
-    log "untainting control-plane nodes"
-    mapfile -t cp_nodes < <(kubectl get nodes | grep control-plane | awk '{print $1}')
-    kubectl taint nodes "${cp_nodes[@]}" node-role.kubernetes.io/control-plane:NoSchedule- || true
-
-    log "tainting windows nodes"
-    mapfile -t windows_nodes < <(kubectl get nodes -o wide | grep Windows | awk '{print $1}')
-    kubectl taint nodes "${windows_nodes[@]}" os=windows:NoSchedule
-
-    log "installing cer-manager"
-    kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.11.0/cert-manager.yaml
-
-    log "wait for cert-manager pods to start"
-    timeout 5m kubectl wait --for=condition=ready pod --all -n cert-manager --timeout -1s
-
-    log "installing admission controller webhook"
-    kubectl apply -f "${SCRIPT_ROOT}/../helpers/hyper-v-mutating-webhook/deployment.yaml"
-
-    log "wait for webhook pods to go start"
-    timeout 5m kubectl wait --for=condition=ready pod --all -n hyperv-webhook-system  --timeout -1s
-
-    log "untainting Windows agent nodes"
-    kubectl taint nodes "${windows_nodes[@]}" os=windows:NoSchedule-
-
-    log "taining master nodes again"
-    kubectl taint nodes "${cp_nodes[@]}" node-role.kubernetes.io/control-plane:NoSchedule || true
-
-    log "done configuring testing for hyperv isolated containers"
-    set +x
-}
-
-run_e2e_test() {
+run_e2e_test_in_pod(){
```

Comment:
This script is primarily used by upstream e2e tests which will run `run_e2e_test` and not `run_e2e_test_in_pod` can we keep most of the functions for the main use case in this script and move the functions related to running the tests in a pod to a different file?

Reply:
Yes, I will move run_e2e_test_in_pod into a separate file